src/main/java/org/aszjch/demoapp/domain/article/ArticleServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+package org.aszjch.demoapp.domain.article;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.aszjch.demoapp.domain.article.port.ArticlePublisher;
+import org.aszjch.demoapp.domain.article.port.ArticleRepository;
+import org.aszjch.demoapp.domain.articlefile.ArticleFile;
+import org.aszjch.demoapp.domain.articlefile.port.ArticleFileAssignmentService;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+
+@Slf4j
+@RequiredArgsConstructor
+public class ArticleServiceImpl implements ArticleService, ArticleFileAssignmentService {
+
+    private final ArticleRepository repository;
+    private final ArticlePublisher publisher;
+
+    @Override
+    public List<Article> get() {
+        return repository.findAll();
+    }
+
+    @Override
+    public Article create(Article article) {
+        Article saved = repository.save(article);
+        publisher.publish(
+                String.valueOf(new ArticleCreatedDto(saved.getId(), saved.getTitle(), saved.getAuthor(),
+                                                     saved.getAbstractText())));
+        return saved;
+    }
+
+    @Override
+    public Optional<Article> getById(Long id) {
+        return repository.findById(id);
+    }
+
+    @Override
+    public Article update(Long id, Article article) {
+        Article entity = getById(id).orElse(new Article());
+        entity.setId(id);
+        entity.setTitle(article.getTitle());
+        entity.setAbstractText(article.getAbstractText());
+        entity.setCreationDate(article.getCreationDate());
+        entity.setAuthor(article.getAuthor());
+        entity.setAbstractText(article.getAbstractText());
+        return repository.save(entity);
+    }
+
+    @Override
+    public void delete(Long id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public void assignFile(ArticleFile articleFile) {
+        Optional<Article> oEntity = getById(articleFile.getArticleId());
+        if (oEntity.isPresent()) {
```

Comment:
Panie Panie. oEntity nic mi nie mówi jako nazwa zmiennej. Chyba taką standardową praktyką jest dodawanie postfixu "Opt" więc byłoby articleOpt.
Druga sprawa: optionale zostały zrobione nie po to, żeby walić ify w każdym miejscu, chociaż to wygodne. Pobawmy się trochę Javą :) i zrób to ładnie w streamie Optional'a - .ifPresent() metoda albo .ifPresentOrElse()